src/utils/isHydrate.js

```diff
@@ -7,4 +7,3 @@ export default (req): boolean => {
   return universal && !isServer() && req.getKind() === 'load'
 }
 
-
```

Comment:
quick note: dont do too much formatting of lines. prettier is currently off. we're gonna go to prettier with line lengths of 86 (the year i was born). Those extract 6 chars make a big difference when u've been stuck at 80 chars for a long time, and it's kind of refreshing. But ultimately i have a coding style which is far different than the mainstream JS/React style of just infinite logical/conditional nesting and braces. The style revolves around assignments. So you will see lots of functions like this:

```js
const func = (arg, arg2) => {
   const somethingNew = arg + arg2
   const bla = 1000
   const somethingElse = somethingNew * bla
  arg = arg || 'default'
  arg2 = arg2 === undefined ? 'default2' : arg2

   // now start doing stuff
}
```

It's somewhat like u'll see in haskell and clojurescript where ur confined to making assignments and instantly using them. But in this case, I just love the readability. I wont type it out, but imagine above with if else blocks and braces in between each assignment--you lose the context, which is that this is the "setup phase." 

The biggest example of this is `src/createRouter.js`. But u'll see it everywhere.

So sometimes these assignments can get long, and just a few extra chars make the difference.

In short, i dont buy the mainstream wisdom "just use prettier" and forget about it. Forget that, reviewing code is time consuming. There's a method to the madness:

## setup phase
- pure functions
- const assignments
- some conditional crap if necessary

## the work phase
- calling some imported functions

----
On top of that, grouping is very important to me/us. Mainstream open source doesnt focus on grouping. By grouping i mean, for example, adding a line break after a bunch of const assignments. 

Sometimes grouping gets ambiguous when there is like a 6 line function, and you end up putting spaces between everything, when each piece of code is a different kinda thing. Its all good. Space is key to visibility. I've still yet to perfect how i do this.

----

Other rules, i approve of one line conditional returns like:

`if (bla) return`

and sometimes i take that a little bit too far where the line gets long. 

But bottom line: i dont like to lose 3 lines for a friggin conditional whose variable is 5 characters. "less code is better code" because it's quicker to grok. If one line does too much, u fuck that up. So it's a balance. BUT if ur doing a lot of similar stuff, those lines can become longer cuz u already know what it's doing.

Another thing: ideally each file/module requires no scrolling to see the entire thing. And ideally its one function. These are ideals obviously. 

----

anyway, just make sure not to go nuts on formatting. i gotta do that myself when i like how things are prior to that stage. Then i gotta format it, and make the final decisions there. You may read code far faster than me, but Ive been having a hard time just remembering everything i wrote. I'm almost back in the zone. And when that point is reached i'll finalize the formatting, turn prettier back on etc. I just gotta make the change myself so i feel comfortable with the transition. Once it reaches that point, this thing is gonna be very much set it and forget it for me. I know that sounds crazy, but i've been OCD the whole way through. There's a million tests, and the way they work is standardized--it wont be hard to fix incoming bugs. So there's just a few final features; then we format it; then i can say "i did it! im happy with it; if other developers like yourself gotta do what they gotta do, im fine with it." That's what im saying. Im very much a waterfall developer. Im too OCD to be any other way, and thats why i do open source rather than apps for clients/employers. We just have the luxury to do that here. In fact, we basically HAVE to be that way since our ware is used by so many other apps; whereas apps can cut a lot more corners. Feel me. That's why app development for other people and their deadlines is just a bad impedance match for me. It's a luxury cuz u gotta find a way to get paid somehow. Lets just put it this way: ive hung on by a thread for several years until recent months. All so i can do whatever the fuck i want! lol. It's like going to Leonardo Da Vinci and rushing him to meet a deadline and him not getting it to the point he's reached the sweet spot where he's happy with it. I wont code any other way anymore. I dont know what luxuries working inside companies provides, but im sure it varies, and im sure it leans like contracting to being forced to released code ur not happy with that works. The future we're laying out is one where we're always happy with whats shipped or we dont fucking ship--cuz we're not desperate for $ or have to please bosses who are pressured by people who are. See what im saying--it all boils down to being pressured for money. What we're trying to create is a world where money is never a burden on us, yet we ship great things and continue to make money all the same.

Reply:
Of course, I only ran eslint —fix

Too time consuming to do cleanups at such a stage, especially since refactoring and organization updates might happen 

When it comes to spaces getting trimmed off files, my IDE does some cleanup built in on save.